extension.ts

```diff
@@ -0,0 +1,34 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * This is the external face of extension.js, the main webpack bundle for the extension.
+ * Anything needing to be exposed outside of the extension sources must be exported from here, because
+ * everything else will be in private modules in extension.js.
+ */
+
+// Export activate/deactivate for entrypoint.js
+export { activateInternal, deactivateInternal } from './src/cosmosExtension';
+
+// Exports for tests
+// The tests are not packaged with the webpack bundle and therefore only have access to code exported from this file.
+//
+// The tests should import '../extension.ts'. At design-time they live in tests/ and so will pick up this file (extension.ts).
+// At runtime the tests live in dist/tests and will therefore pick up the main webpack bundle at dist/extension.js.
+export { ext } from './src/extensionVariables';
+export * from './src/utils/array';
+export { AttachedAccountsTreeItem, MONGO_CONNECTION_EXPECTED } from './src/tree/AttachedAccountsTreeItem';
+export { parseDocDBConnectionString } from './src/docdb/docDBConnectionStrings';
+export { emulatorPassword } from './src/constants';
+export { getDocumentTreeItemLabel } from './src/utils/vscodeUtils';
+export { validOnTimeoutOrException } from './src/utils/inputValidation';
+export { addDatabaseToAccountConnectionString, getDatabaseNameFromConnectionString } from './src/mongo/mongoConnectionStrings';
+export { MongoCommand } from './src/mongo/MongoCommand';
+export { getAllCommandsFromText, getCommandFromTextAtLocation } from './src/mongo/MongoScrapbook';
+export { rejectOnTimeout, valueOnTimeout } from './src/utils/timeout';
+
+// These use instanceof and therefore we need to make sure we're using the same version of the bson module in the tests as in the bundle,
```

Comment:
Does this only apply to test code? If not, it makes me kinda nervous. What exactly is the problem with instanceof? I guess I'd rather find a fix for instanceof than single out every single use of it we run in to.

Is this at all related to the thing you did here? https://github.com/Microsoft/vscode-docker/pull/745 Either way don't forget to do that here :)

Reply:
Yes, tests only (updated comment).  The issue is that the test code is completely separate from the extension bundle, so if we do a blind import from the tests, you get two different instances of the module and hence two difference classes with different identity.  That's also why we're importing everything from extension.ts instead of importing individual files directly from the tests like we used to, because you'd end up with two different instances of the code (and hence extensionVariables isn't a singleton, etc.).  

extension.js is separate from the test code and uses an internal webpack version of import, so it doesn't know about anything imported by code outside of it (something I wish we could change - I tried).

Reply:
> Is this at all related to the thing you did here? Microsoft/vscode-docker#745 Either way don't forget to do that here :)

No.  The other fix will be in the shared version.